### DISCUSSION TITLE
Why subtracting the max log posterior when plotting the walks?

### DISCUSSION THREAD

**Discussion by snwnde** [Samreay/ChainConsumer#110] [Q&A] [answered]
I was surprised by the results when plotting the log posterior by calling `plot_walks`, until I found,
https://github.com/Samreay/ChainConsumer/blob/50fd63b15df93c87a918e8b05df5143119712fbf/src/chainconsumer/plotter.py#L382
Could someone please explain the rationale behind this subtraction by the maximal log posterior?

**Comment by Samreay** [comment, marked as answer]
I am happy to remove this, but this was added back in the day where we only fed chains from emcee and similar samplers into chainconsumer, and those samplers only sample proportional to the posterior, not the actual posterior (pesky evidence on the denominator ruining our lives). As such, it made sense to substract the maximum value to get a normalised view of the posterior samples

**Reply by snwnde** [reply]
Thanks a lot for the anwser. I agree our so-called "posterior" is ususally only proportional to the actual one. I would suggest adding an option `normalised` default to False (or True ?) but I am curious what the other users think about it.

**Reply by Samreay** [reply]
Hmm, I feel like we're not going to get more input that just the two of us on this one. My question would be whether adding the parameter does serve a purpose in the output plots. Its only in the walks plot, right, which is a diagnostic plot?

**Reply by snwnde** [reply]
Apologies for the delayed response. I took the time to further explore and consider my use case. 

Adding this parameter would not have any influence on the output plots. Also the walks plot is unlikely to be included in the final publication as it is created solely for diagnostic purposes as you pointed out. 

There are situations where it makes sense to plot the un-normalised "log-posteriors". For instance, an additional deterministic signal in the data almost surely has an impact on the likelihood function and hereby the un-normalised "log-posteriors". This information is useful for the diagnostic but it would be killed by the normalisation.
